operator/v1/types_machineconfiguration.go

```diff
@@ -153,3 +167,181 @@ const (
 	// This feature only supports MAPI machinesets at this time.
 	MachineAPI MachineManagerMachineSetsAPIGroupType = "machine.openshift.io"
 )
+
+type NodeDisruptionPolicySpec struct {
+	// userPolicies define user-provided node disruption policies
+	// +optional
+	UserPolicies NodeDisruptionPolicyConfig `json:"userPolicies"`
```

Comment:
Off the top of my head any potential future extensions would be with the new image based workflow. I think an example could be I could specify some additional non-machineconfig content (image based) and specify something here to indicate how it can be applied.

But maybe for now we can just collapse this to the higher level policy and add a new field when that comes into play?